lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -467,4 +421,80 @@ private boolean isH2OrPostgresIntegrityConstraintViolation(PersistenceException
+     * @param classificationImpl the classification
+     * @return the old classification
+     */
+    private Classification checkIfClassificationBasedNewest(ClassificationImpl classificationImpl)
+        throws ConcurrencyException, ClassificationNotFoundException {
+        Classification oldClassification = this.getClassification(classificationImpl.getKey(),
+            classificationImpl.getDomain());
+        if (!oldClassification.getModified().equals(classificationImpl.getModified())) {
+            throw new ConcurrencyException(
+                "The current Classification has been modified while editing. The values can not be updated. Classification "
+                    + classificationImpl.toString());
+        }
+        return oldClassification;
+    }
+
+    /**
+     * Update classification fields used by tasks.
+     * @param classificationImpl the new classification
+     * @param oldClassification the old classification
+     */
+    private void updateClassificationFieldUsedByTask(ClassificationImpl classificationImpl, Classification oldClassification) {
+        List<TaskSummary> taskSummaries = taskanaEngine.getTaskService()
+            .createTaskQuery()
+            .classificationIdIn(oldClassification.getId())
+            .list();
+
+        if (!taskSummaries.isEmpty()) {
+            List<String> taskIds = new ArrayList<>();
+            taskSummaries.forEach(ts -> taskIds.add(ts.getTaskId()));
+            taskMapper.updateClassificationCategoryOnChange(taskIds, classificationImpl.getCategory());
+        }
+    }
+
+    /**
+     * Check if parentId or parentKey were changed and if the classification exist.
+     * @param classificationImpl the new classification
+     * @param oldClassification the old classification
+     * @throws ClassificationNotFoundException
+     */
+    private void checkIfParentIdOrParentKeyChangedAndIfExist(Classification oldClassification, ClassificationImpl classificationImpl)
+        throws ClassificationNotFoundException {
+        if (!oldClassification.getParentId().equals(classificationImpl.getParentId())
+            && classificationImpl.getParentId() != null && !classificationImpl.getParentId().isEmpty()) {
+                this.getClassification(classificationImpl.getParentId());
+        }
+
+        if (!oldClassification.getParentKey().equals(classificationImpl.getParentKey())
+            && classificationImpl.getParentKey() != null && !classificationImpl.getParentKey().isEmpty()) {
+                this.getClassification(classificationImpl.getParentKey(), classificationImpl.getDomain());
+        }
+    }
+
+    /**
+     * Check if priority or service level were changed and create the new job.
+     * @param classificationImpl the new classification
+     * @param oldClassification the old classification
+     */
+    private void checkIfPriorityOrServiceLevelChangedAndCreateJob(Classification oldClassification, ClassificationImpl classificationImpl) {
```

Comment:
createJobIfPriorityOrServiceLevelHasChanged ?

Reply:
This name is perfect. Thanks, Holger!